### DISCUSSION TITLE
hook for Background Fetching Loading States similar to useIsFetching

### DISCUSSION THREAD

**Discussion by FranciscoGileno** [TanStack/query#1062] [General]
Thanks for the awesome crazy lib.

I noticed there is a hook called `useIsFetching` which gives you the number of requests being fetched.

I wonder if there is a way to have a simular hook only with the requests that are running in backend([Displaying Background Fetching Loading States](https://react-query.tanstack.com/docs/guides/queries#displaying-background-fetching-loading-states)).

I just had a look at the `useIsFetching` [code](https://github.com/tannerlinsley/react-query/blob/master/src/react/useIsFetching.ts), and it is pretty straight forward... I mean something like this:

```js
import React from 'react'

import { useQueryCache } from './ReactQueryCacheProvider'
import { useIsMounted } from './utils'

export function useIsBackgroundFetching(): number {
  const isMounted = useIsMounted()
  const queryCache = useQueryCache()
  const [isBackgroundFetching, setIsBackgroundFetching] = React.useState(queryCache.isFetching && !queryCache.isLoading)

  React.useEffect(
    () =>
      queryCache.subscribe(() => {
        if (isMounted()) {
          setIsBackgroundFetching(queryCache.isFetching && !queryCache.isLoading)
        }
      }),
    [queryCache, setIsBackgroundFetching, isMounted]
  )

  return isBackgroundFetching
}
```

I can't do this on my project since `react-query` does not export `useIsMounted`.

**Comment by StasSemak** [comment]
I run into similar issue, using `react-query` v5, so I created a workaround based on original `useIsFetching` [code](https://github.com/TanStack/query/blob/main/packages/react-query/src/useIsFetching.ts) to track only background fetching. Here it is:
```js
import { QueryClient, QueryFilters, notifyManager, useQueryClient } from "@tanstack/react-query";
import React from "react";

export function useIsBgFetching(
  filters?: QueryFilters,
  queryClient?: QueryClient,
): number {
  const client = useQueryClient(queryClient)
  const queryCache = client.getQueryCache()

  return React.useSyncExternalStore(
    React.useCallback(
      (onStoreChange) =>
        queryCache.subscribe(notifyManager.batchCalls(onStoreChange)),
      [queryCache],
    ),
    () => client.getQueryState(filters?.queryKey ?? [""])?.status !== "pending" ? client.isFetching(filters) : 0,
    () => client.getQueryState(filters?.queryKey ?? [""])?.status !== "pending" ? client.isFetching(filters) : 0,
  )
}
```

Might be not the best or the most optimal solution, but works fine in my case, and may be useful for somebody. 
Also, worth mentioning that this approach is focused on tracking only one query and not useful if you need info about all queries in your app.

**Reply by TkDodo** [reply]
FYI, `useIsFetching` already takes filters, and you can narrow this to `status: 'pending'`:

```
useIsFetching({
  // if we have data and are fetching, it means a background refetch
  predicate: (query) => query.state.data !== undefined
})
```

**Reply by StasSemak** [reply]
Oh, thanks, this would be so much better solution!
